### PR TITLE
added a few methods to ListingCart

### PR DESCRIPTION
**Pull request by ryanhertz**
@rich @baron-hines @bhornseth @mjobrien

### CODE REVIEW

File: lib/spark_api/models/listing_cart.rb [hunk 1]
```diff
@@ -52,6 +63,10 @@ def self.portal(arguments={})
           collect(connection.get("/#{self.element_name}/portal", arguments))
       end
 
+      def deletable?
+        !attributes.has_key?("PortalCartType")
```
Comment (ryanhertz) [inline lib/spark_api/models/listing_cart.rb]: I'm not positive this logic is correct. This should be true for custom carts, and false for vow carts like "Favorites". Custom carts don't appear to have this attribute, but I would be more comfortable if the attribute were more closely matched to the purpose.

File: spec/unit/spark_api/models/listing_cart_spec.rb [hunk 2]
```diff
@@ -133,4 +133,39 @@
     end
   end
 
+  describe "#listings" do
```
Comment (ryanhertz) [inline spec/unit/spark_api/models/listing_cart_spec.rb]: The other tests in this file use this format:
```
context "/listingcarts/<cart_id>", :support do
  on_get_it "should get a listing cart" do
    ...
  end
end
```
The way I've done it makes more sense to me, but if some can explain why it should be done like this example, I'd be happy to change it.

File: lib/spark_api/models/listing_cart.rb (hunk 1 above)
Comment (GutsyGusty) [inline lib/spark_api/models/listing_cart.rb]: I'm pretty sure custom carts should have a PortalCartType of "Custom"

Comment (ryanhertz) [inline lib/spark_api/models/listing_cart.rb]: My test custom cart, created by a private user looks like this:
```
#<SparkApi::Models::ListingCart:0x007fbf8c943c90 @attributes={
  "ResourceUri"=>"/v1/listingcarts/20150412180935247711000000", 
  "Id"=>"20150412180935247711000000", 
  "Name"=>"a big collection", 
  "ListingCount"=>200
}, @errors=[]>
```

**Pull request merged**